pkg/tfbridge/provider.go

```diff
@@ -187,48 +187,46 @@ func (p *Provider) Configure(ctx context.Context, req *pulumirpc.ConfigureReques
 		}
 	}
 
+	// First make a Terraform config map out of the variables. We do this before checking for missing properties
+	// s.t. we can pull any defaults out of the TF schema.
+	config, err := MakeTerraformConfig(nil, vars, p.tf.Schema, p.info.Config, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal config state")
+	}
+
+	if p.info.PreConfigureCallback != nil {
+		if err = p.info.PreConfigureCallback(vars, config); err != nil {
+			return nil, err
+		}
+	}
+
 	// So we can provide better error messages, do a quick scan of required configs for this
 	// schema and report any that haven't been supplied.
 	var missingKeys []*pulumirpc.ConfigureErrorMissingKeys_MissingKey
 	for key, meta := range p.tf.Schema {
-		_, present := vars[resource.PropertyKey(key)]
-		if meta.Required && !present {
-			fullyQualifiedKey := tokens.NewModuleToken(p.pkg(), tokens.ModuleName(key))
+		if meta.Required && !config.IsSet(key) {
+			name := TerraformToPulumiName(key, meta, false)
```

Comment:
what's the difference between a terraform and pulumi name here?

Reply:
Terraform names `look_like_this`; Pulumi names `lookLikeThis`. When we iterate over the TF schema map we get TF names, but we want to report Pulumi names in the error.

Reply:
gotcha, thanks!